### PR TITLE
Df 58 prompt engineering + prmptingService

### PR DESCRIPTION
**Pull request by jonkra20**
# DQL-\<NUM\>: \<TITLE\>
`<DESCRIPTION>`

## :white_check_mark: Pre-Merge Checklist
All boxes must be checked before the Pull Request is merged. This checklist has to be completed by the reviewer.

- [X] The CI pipeline has passed (this is enforced by GitHub)
- [x] Code Review
    - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR implements what is described in the JIRA-Issue
- [x] The new feature has been tested by the reviewer on his local machine
- [x] A deployment to the `dev` environment has succeeded. If possible, ensure that the feature can be observed on the server.
- [x] If required, any documentation is updated accordingly

## :floppy_disk: Merge Checklist
- [x] The branch is merged to main using the "Squash and Merge" strategy following the commit message naming convention `DQL-<NUM>: <TITLE>`
- [x] The branch is deleted after the PR is merged

### CODE REVIEW

File: src/dataland_qa_lab/dataland/data_extraction.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import pypdf
+from azure.ai.documentintelligence import DocumentIntelligenceClient
+from azure.ai.documentintelligence.models import AnalyzeResult, ContentFormat
+from azure.core.credentials import AzureKeyCredential
+from openai import AzureOpenAI
+
+from dataland_qa_lab.utils import config
+
+
+def get_relevant_page_of_pdf(page: int, full_pdf: pypdf.PdfReader) -> io.BytesIO:
```
Comment (TilmanNiem) [inline src/dataland_qa_lab/dataland/data_extraction.py:12]: warum benutzt ihr wieder die alte data extraction klasse statt der neuen?

Comment (jonkra20) [reply src/dataland_qa_lab/dataland/data_extraction.py:12]: Ich habe mich auch schon gewundert. Die kann ich löschen oder die existiert ja nicht mehr? Wie heißt die neue Klasse?

Comment (TilmanNiem) [reply src/dataland_qa_lab/dataland/data_extraction.py]: der yes-no-value-generator

**Review by fschnizer — APPROVED** [review]
LGTM! 👍

**Review by Opos03 — APPROVED** [review]
LGTM 🐐

**Pull request merged**